### PR TITLE
Refactor registries, and added a registry for biomes

### PR DESCRIPTION
**Pull request by me4502**
This removes a lot of adapter code from Bukkit and Sponge, and also just overly simplifies how biomes work in WorldEdit.

### CODE REVIEW

File: worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitAdapter.java [hunk 1]
```diff
@@ -304,6 +306,27 @@ public static GameMode adapt(org.bukkit.GameMode gameMode) {
...
+    }
+
+    public static Biome adapt(BiomeType biomeType) {
+        if (!biomeType.getId().startsWith("minecraft:")) {
+            throw new IllegalArgumentException("Bukkit only supports vanilla biomes");
+        }
+        try {
+            return Biome.valueOf(biomeType.getId().substring(10).toUpperCase());
+        } catch (IllegalArgumentException e) {
+            return null;
```
Comment (kenzierocks) [inline worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitAdapter.java]: This is weird -- either we should return `null` in all error cases, or throw an error. You do the former here, and the latter above.

Comment (me4502) [reply worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitAdapter.java]: This is the only one using the Enum#valueOf method, which throws an exception. I don't believe the others do?

Comment (kenzierocks) [reply worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitAdapter.java]: Oh, weird. Looks like it'd be consistent then.

File: worldedit-core/src/main/java/com/sk89q/worldedit/LocalConfiguration.java [hunk 2]
```diff
@@ -138,6 +77,73 @@
...
+                BlockTypes.DARK_OAK_BUTTON,
+                BlockTypes.JUNGLE_BUTTON,
+                BlockTypes.OAK_BUTTON,
+                BlockTypes.SPRUCE_BUTTON,
+                BlockTypes.CACTUS,
+                BlockTypes.SUGAR_CANE,
+                // ores and stuff
+                BlockTypes.BEDROCK
+        );
+        return blockTypes.stream().filter(type -> !Objects.isNull(type)).map(BlockType::getId).toArray(String[]::new);
```
Comment (kenzierocks) [inline worldedit-core/src/main/java/com/sk89q/worldedit/LocalConfiguration.java]: 95% sure there's a `isNotNull`.

**Comment by me4502** [comment]
So realistically yes.

I forgot weather existed, so I’ll do that. As for fluids, we can’t populate it from Spigot as they don’t have a system for it. I assume once it actually gets used within the game they’ll add one.

As for the categories... getting that to generate in MCUtils is fairly complex as they’re loaded from JSON files, and it’s not a clean process.

**Comment by wizjany** [comment]
> As for the categories... getting that to generate in MCUtils is fairly complex as they’re loaded from JSON files, and it’s not a clean process.

Ah yea, forgot about data packs entirely.

**Pull request merged**